cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -608,6 +609,13 @@ private void setStreamEventDecoder(Job job) throws IOException {
     }
   }
 
+  private String getJobName(BasicMapReduceContext context) {
+    Id.Program programId = context.getProgram().getId();
+    return String.format("%s.%s.%s.%s.%s",
```

Comment:
added a comment.
And will work on an audi test for all this (since its not a usable feature in Standalone/InMemory).